server_test.go

```diff
@@ -208,6 +208,51 @@ func TestOpenStatRace(t *testing.T) {
 	checkServerAllocator(t, server)
 }
 
+func TestOpenWithPermissions(t *testing.T) {
+	client, server := clientServerPair(t)
+	defer client.Close()
+	defer server.Close()
+
+	tmppath := path.Join(os.TempDir(), "open_permissions")
+	pflags := flags(os.O_RDWR | os.O_CREATE | os.O_TRUNC)
+	ch := make(chan result, 2)
+	id1 := client.nextID()
+	id2 := client.nextID()
+
+	// New files should have their permissions set.
+	client.dispatchRequest(ch, &sshFxpOpenPacket{
+		ID:     id1,
+		Path:   tmppath,
+		Pflags: pflags,
+		Flags:  sshFileXferAttrPermissions,
+		Attrs:  []byte{0x0, 0x0, 0x1, 0xe5}, // 0o745 -- a slightly strange permission to test.
+	})
+	<-ch
+	stat, err := os.Stat(tmppath)
+	assert.NoError(t, err)
```

Comment:
This needs to be a `require.NoError` as if `err != nil` then `stat == nil` and the following line will panic.